### DISCUSSION TITLE
The Tape Tool (Type: Rectangular) and and automatically created vectors (gif).

### DISCUSSION THREAD

**Discussion by beeheemooth** [opentoonz/opentoonz#6775] [General]
Auto-generated vectors are not updated to determine gaps between them.

### Steps to Reproduce

1. Using Vector Brush in Frame Range mode, draw an animation of several non-intersecting lines. You can also create auto inbetweens in the usual way (copy-paste vectors from the first frame to the last frame + inbetweens).
![ezgif-1-67980cb98f10](https://user-images.githubusercontent.com/25724133/52867984-d86eed00-314a-11e9-83ee-4394240b76bd.gif)

2. Turn on Gap Check. Please note that lines that were created are not automatically recognized in Gap Check mode.
![ezgif-1-c54bbe35b1ed](https://user-images.githubusercontent.com/25724133/52868030-f50b2500-314a-11e9-825f-8b895e335a1f.gif)

3. Using the Tape Tool Type: Rectangular, try to connect the ends of the vectors. This can only be done in the first and last drawing.
![ezgif-1-d4ea77979285](https://user-images.githubusercontent.com/25724133/52868161-3b608400-314b-11e9-9706-ad75bf24f37e.gif)

4. Select the Control Point Editor Tool and highlight in turn all the automatically created lines in each frame. As soon as you do this, Gap Check will start working and its lines will be displayed.
![ezgif-1-f0e6a2adf016](https://user-images.githubusercontent.com/25724133/52868192-4ca99080-314b-11e9-885b-2b09cfd1be94.gif)

5. Now the Tape Tool will work properly.
![ezgif-1-27978bdd74ac](https://user-images.githubusercontent.com/25724133/52868212-5e8b3380-314b-11e9-847e-1a2fc572aa40.gif)

### Expected Results

Ability to apply the Tape Tool Type Rectangular to vector lines created automatically.
### Actual Results

The Vector Tape Tool (Rectangular) cannot be applied to vector lines created automatically.
### System Information
* **OpenToonz Version**: v1.3
* **Operating System**: Windows 8.1

**Comment by WolfInABowl** [comment]
I can reproduce this.
Version: 1.3
OS: Windows 10

**Comment by andeon** [comment]
@Abel032 I haven’t checked if your gap improvements affect the vector side at all, but this specific issue still persists. I’m not sure if it’s really related to the gap feature.

Following **step 4** shared by the OP works, so it might be related to some update in the vector points.

For example, using the Control Point Editor tool, if you click on 2 extreme points and change the frames, the gap appears and the tool works normally. However, this needs to be done for all points.

I’m not sure if this is a common workflow, since creating spaced points just to automatically generate inbetweens and then use the Tape Tool seems counterproductive. Still, the fact that vectors don’t have their initial points properly marked appears to be a problem.
